src/middlewared/middlewared/schema.py

```diff
@@ -957,16 +957,12 @@ def nf(*args, **kwargs):
                 args, kwargs = clean_and_validate_args(args, kwargs)
                 return f(*args, **kwargs)
 
-        nf.__name__ = f.__name__
-        nf.__doc__ = f.__doc__
-        # Copy private attrs to new function so decorators can work on top of it
-        # e.g. _pass_app
-        for i in dir(f):
-            if i.startswith('__'):
-                continue
-            if i.startswith('_'):
-                setattr(nf, i, getattr(f, i))
+        from middlewared.utils.type import copy_function_metadata
```

Comment:
Does this really need to be lazy imported?

Reply:
Yes, otherwise it would be circular dependency to break which we'll need to move stuff out of utils/__init__.py